drivers/input/input_gt911.c

```diff
@@ -133,26 +132,32 @@ static int gt911_process(const struct device *dev)
 		return r;
 	}
 
-	/* obtain first point X, Y coordinates and event from:
-	 * REG_P1_XH, REG_P1_XL, REG_P1_YH, REG_P1_YL.
+	/*
+	 * obtain point X, Y coordinates and event
 	 */
-	reg_addr = REG_FIRST_POINT;
-	r = gt911_i2c_write_read(dev, &reg_addr, sizeof(reg_addr),
-				 &pointRegs, sizeof(pointRegs));
-	if (r < 0) {
-		return r;
-	}
-
-	pressed = (points == 1);
-	row = ((pointRegs.highY) << 8U) | pointRegs.lowY;
-	col = ((pointRegs.highX) << 8U) | pointRegs.lowX;
-
-	LOG_DBG("pressed: %d, row: %d, col: %d", pressed, row, col);
-
-	if (pressed) {
-		input_report_abs(dev, INPUT_ABS_X, col, false, K_FOREVER);
-		input_report_abs(dev, INPUT_ABS_Y, row, false, K_FOREVER);
-		input_report_key(dev, INPUT_BTN_TOUCH, 1, true, K_FOREVER);
+	if (points) {
+#ifdef CONFIG_INPUT_GT911_MULTITOUCH_ENABLED
+		if (points > 1) {
+			input_report_abs(dev, INPUT_ABS_MT_SLOT, points, true, K_FOREVER);
+		}
```

Comment:
Wouldn't this result in the MT slot being only generated for points 2 and beyond? Then the application would not be able to identify point 1 anymore (also it should probaby start from 0). I think this should always be sent if multitouch support is enabled (I would squash the last two patches to make things easier to review) and this should be indexed by `points -1`, does that make sense?

Reply:
MT slot is generated with every group of points exceeding 1. (I hope I understood what you meant) 
It makes sense to me to only send a Multitouch event when there is a multitouch AND multitouch is enabled. However, it doesn't affect the way the driver recognizes the points. When MT is disabled, the driver only takes into consideration the first point. (the same previous behavior)
![image](https://github.com/zephyrproject-rtos/zephyr/assets/166378078/e22cc16e-5f88-421e-95ab-ce716f87a9dd)
 Conversely, we'll have the MT event with the number of touches + the points (indexed from 0 to points - 1) 
![image](https://github.com/zephyrproject-rtos/zephyr/assets/166378078/a8df5ac2-d33f-4b0c-8272-d3c297d26e58)
I suggest we add a message after "Input sample started" stating whether MT is enabled or not.

Reply:
`INPUT_ABS_MT_SLOT` should indicate what slot the points that is updated refers to, I think as it's written now it indicates the number of points, which does not really work out, this should be part of the cycle below, every points should be reported as a sequence of
- MT slot
- X
- Y
- touch

Reply:
Now I get it. I didn't quite understand the meaning of INPUT_ABS_MT_SLOT from the [documentation](https://docs.zephyrproject.org/latest/services/input/index.html#c.INPUT_ABS_MT_SLOT). I did the modifications and the sequence of events is as you said: 
![image](https://github.com/zephyrproject-rtos/zephyr/assets/166378078/51fbf026-87d6-4fed-a8d8-f50f92b63c36)

Reply:
@fabiobaltieri, is it okay for you now? 
I was also wondering if the release event (INPUT_BTN_TOUCH,0) should be generated when there are no points at all or for each point individually like the other events.
```
			input_report_abs(dev, INPUT_ABS_X, col, false, K_FOREVER);
			input_report_abs(dev, INPUT_ABS_Y, row, false, K_FOREVER);
			input_report_key(dev, INPUT_BTN_TOUCH, 1, true, K_FOREVER);
		}
	} else {
		input_report_key(dev, INPUT_BTN_TOUCH, 0, true, K_FOREVER);
	}
```

Reply:
It should be generated for each individual point

Reply:
Makes sense, done. :)

Reply:
@fabiobaltieri, do you agree with the changes?